libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/MapboxTripService.kt

```diff
@@ -62,17 +64,24 @@ internal class MapboxTripService(
         tripNotification,
         {
             try {
-                applicationContext.startService(intent)
-            } catch (e: IllegalStateException) {
                 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                     applicationContext.startForegroundService(intent)
                 } else {
-                    throw e
+                    applicationContext.startService(intent)
                 }
+                true
+            } catch (ex: Throwable) {
+                logE("Foreground service could not be started: $ex. Make sure you invoke " +
+                    "`MapboxNavigation#startTripSession` when the app is in foreground.")
```

Comment:
Being not in foreground isn't the only reason when a throwable might be thrown. 
<img width="615" alt="image" src="https://user-images.githubusercontent.com/6190346/193522331-db22c89b-35c3-49af-88e9-61e7e149abdd.png">
Maybe it's better to rely on exception for explanation? 
```suggestion
                logE("Foreground service could not be started: $ex.")
```

Reply:
I've included the exception in my log. Invoking from background will be the most frequent case I guess. It's not the only reason but it is something the user should first look into. Changed the description a bit.